src/Reflection/ReflectionProperty.php

```diff
@@ -207,7 +207,7 @@ public function isInitialized(object|null $object = null): bool
 
     public function isReadOnly(): bool
     {
-        return $this->node->isReadonly();
+        return $this->getDeclaringClass()->isReadOnly() || $this->node->isReadonly();
```

Comment:
Minor optimization possible by reversing the operand order?

Reply:
Done.